cubeb-backend/src/log.rs

```diff
@@ -23,8 +23,21 @@ macro_rules! cubeb_log_internal {
     };
     (__INTERNAL__ $msg: expr) => {
         if let Some(log_callback) = $crate::ffi::g_cubeb_log_callback {
-            let filename = std::path::Path::new(file!()).file_name().unwrap().to_str().unwrap();
-            let cstr = ::std::ffi::CString::new(format!("{}:{}: {}\n", filename, line!(), $msg)).unwrap();
+            use std::io::Write;
+
+            let mut buf = [0 as u8; 1024];
+            let filename = std::path::Path::new(file!())
+                .file_name()
+                .unwrap()
+                .to_str()
+                .unwrap();
+            // 2 for ':', 1 for ' ', 1 for '\n', and 1 for converting `line!()` to number of digits
+            let len = filename.len() + ((line!() as f32).log10() as usize) + $msg.len() + 5;
```

Comment:
This needs to be adjusted, it's not quite correct `log10` of 900 is `2.9542425094393248`., we need to round (converting integer always rounds towards zero).

Reply:
I am not sure if I understand the comment correctly. The `len` has a `+1` (one of thr `+5`) for getting the number of digits we need. 
When `line!()` is `900`, the `(line!() as f32).log10()` will retrun `2.9542425` so `(line!() as f32).log10() as usize` is `2`. As a result, the number of digits we need for `900` is `2 + 1`. The digits calculated for `900` is `3` indeed. [Here][link] is a link to test.

To make this more clear, I can use `f32::floor` or `f32::trunc`. The `floor` and `trunc` return the same value when the `f32` is positive. Since `line!()` must be a positive number, so we can use any one of them.

[link]: https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&code=fn%20main()%20%7B%0A%20%20%20%20let%20line%3A%20u32%20%3D%20900%3B%0A%20%20%20%20let%20log_value%20%3D%20(line%20as%20f32).log10()%2F*.trunc()*%2F%3B%0A%20%20%20%20let%20to_usize%20%3D%20log_value%20as%20usize%3B%0A%20%20%20%20let%20digits%20%3D%20to_usize%20%2B%201%3B%0A%20%20%20%20println!(%0A%20%20%20%20%20%20%20%20%22log_value%3A%20%7B%7D%2C%20to_usize%3A%20%7B%7D%2C%20digits%3A%20%7B%7D%22%2C%0A%20%20%20%20%20%20%20%20log_value%2C%20to_usize%2C%20digits%0A%20%20%20%20)%3B%0A%7D%0A